src/sagemaker/jumpstart/factory/estimator.py

```diff
@@ -793,3 +797,73 @@ def _add_fit_extra_kwargs(kwargs: JumpStartEstimatorFitKwargs) -> JumpStartEstim
             setattr(kwargs, key, value)
 
     return kwargs
+
+
+def _add_config_name_to_kwargs(
+    kwargs: JumpStartEstimatorInitKwargs,
+) -> JumpStartEstimatorInitKwargs:
+    """Sets tags in kwargs based on default or override, returns full kwargs."""
+
+    specs = verify_model_region_and_return_specs(
+        model_id=kwargs.model_id,
+        version=kwargs.model_version,
+        scope=JumpStartScriptScope.TRAINING,
+        region=kwargs.region,
+        tolerate_vulnerable_model=kwargs.tolerate_vulnerable_model,
+        tolerate_deprecated_model=kwargs.tolerate_deprecated_model,
+        sagemaker_session=kwargs.sagemaker_session,
+        config_name=kwargs.training_config_name,
+    )
+
+    if kwargs.base_job_name:
+        _, _, _, base_training_config_name = get_model_info_from_training_job(
+            training_job_name=kwargs.base_job_name, sagemaker_session=kwargs.sagemaker_session
+        )
+
+        kwargs.training_config_name = (
+            kwargs.training_config_name
+            or specs.training_configs.configs.get(
+                base_training_config_name
+            ).default_incremental_trainig_config
+            or specs.training_configs.get_top_config_from_ranking().default_incremental_trainig_config  # noqa E501  # pylint: disable=c0301
+        )
+
+    if specs.training_configs and specs.training_configs.get_top_config_from_ranking().config_name:
+        kwargs.training_config_name = (
+            kwargs.training_config_name
+            or specs.training_configs.get_top_config_from_ranking().config_name
+        )
+
+        kwargs.inference_config_name = (
```

Comment:
this logic should be in the `model` factory. Can't you call `_get_deploy_kwargs` from the `estimator` factory to get the inference config name?